frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx

```diff
@@ -141,9 +141,15 @@ function VariableItem({
 			}
 	};
 
-	const selectValue = variableData.allSelected
-		? 'ALL'
-		: variableData.selectedValue?.toString() || '';
+	const getSelectValue = (): string | string[] => {
+		if (Array.isArray(variableData.selectedValue)) {
+			return variableData.selectedValue.map((item) => item.toString());
+		}
+		return variableData.selectedValue?.toString() || '';
+	};
+
+	const selectValue = variableData.allSelected ? 'ALL' : getSelectValue();
```

Comment:
only the concern is that when the component is re-rendered function context would be re-created so can we have a memo and a single utility to get the selectedValue

Reply:
okay i've moved the function above and added a memo